libcontainerd/container_windows.go

```diff
@@ -249,7 +249,13 @@ func (ctr *container) waitExit(process *process, isFirstProcessToStart bool) err
 						}
 						logrus.Error(err)
 					} else {
-						ctr.client.Create(ctr.containerID, ctr.ociSpec, ctr.options...)
+						if err := ctr.client.Create(ctr.containerID, ctr.ociSpec, ctr.options...); err != nil {
+							si.State = StateExit
```

Comment:
Sounds like a better flow to move it after the `StateChanged()` called indeed. especially since `<-wait` only blocks until the restart timeout expires which default to `100ms` the first time around if I'm not mistaken.

Also, having the backend registering the status change to `StateRestart` first, just makes sense.